src/engine/threaded_engine.cc

```diff
@@ -279,10 +283,29 @@ void ThreadedEngine::Push(OprHandle op, Context exec_ctx, int priority) {
 void ThreadedEngine::PushAsync(AsyncFn fn, Context exec_ctx,
                                std::vector<VarHandle> const& const_vars,
                                std::vector<VarHandle> const& mutable_vars,
-                               FnProperty prop, int priority) {
-  ThreadedOpr *opr = NewOperator(std::move(fn), const_vars, mutable_vars, prop);
+                               FnProperty prop,
+                               int priority,
+                               bool profiling,
+                               const char* opr_name) {
+  ThreadedOpr *opr = NewOperator(std::move(fn), const_vars, mutable_vars, prop, opr_name);
   opr->temporary = true;
-  Push(opr, exec_ctx, priority);
+  Push(opr, exec_ctx, priority, profiling);
+}
+
+void ThreadedEngine::PushSync(SyncFn exec_fn, Context exec_ctx,
+                              std::vector<VarHandle> const& const_vars,
+                              std::vector<VarHandle> const& mutable_vars,
+                              FnProperty prop,
+                              int priority,
+                              bool profiling,
```

Comment:
good catch. We can remove this argument now because we control whether to record the operator by Profiler::State and Profiler::Mode instead of this argument. And the implementation of PushSync is united again, so we can make it inline.